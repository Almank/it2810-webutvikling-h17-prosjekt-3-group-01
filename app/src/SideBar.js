/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import {LinkButton} from "./components/SideBar/LinkButton";
import { View, StyleSheet } from 'react-native'


export class SideBar extends React.Component {
    render(){
        return (
            <View className="sideBar" style={styles.container}>
                <LinkButton/>
                <LinkButton/>
                <LinkButton/>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        backgroundColor: 'grey',
        flexDirection: 'row',
    },
});