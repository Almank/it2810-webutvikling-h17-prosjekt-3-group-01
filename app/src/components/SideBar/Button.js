/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import {Link} from 'react-router-dom';

export class Button extends React.Component {
    render(){
        return (
            <Link className="Button" to={this.props.link}>
                <span className={'glyphicon ' + this.props.icon} />
            </Link>
        );
    }
}