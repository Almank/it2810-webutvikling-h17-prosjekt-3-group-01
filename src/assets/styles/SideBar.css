.SideBar {
    display: flex;
    width: 100px;
    height: 100%;
    background-color: #35393a;
    flex-direction: column;
    align-items: center;
}

.SideBar .Button{
    width: 80px;
    height: 80px;
    background-color: #17292d;
    margin-top: 5px;
    color: white;
    text-decoration: none;
    padding: 5px;
}

.Glyphicon{
    color: white;
    font-size: 30px;
    text-align: center;
    padding-top: 20px;
    padding-left: 20px;
}

.Glyphicon:hover .Button{
    color: #ECA400;
}