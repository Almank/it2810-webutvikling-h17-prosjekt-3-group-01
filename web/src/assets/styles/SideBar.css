.sideBar {
    display: flex;
    min-height: 100vh;
    height: auto;
    background-color: #35393a;
    flex-direction: column;
}

.sideBar .Button{
    width: 80px;
    height: 80px;
    display: flex;
    align-items: center;
    justify-content: center;
}

.Button span{
    color: white;
    font-size: 42px;
}

.Button:hover .glyphicon{
    color: #FF9505;
}
.Button:active .glyphicon{
    color: #FF9505;
}

.sideBar .Button:active{
    color: #FF9505;
}

.Button:hover, .Button  {
    text-decoration: none !important;
}

/* Mobile - portrait */
@media only screen and (max-width : 680px) {
    .sideBar {
        flex-direction: row;
        justify-content: space-between;
        min-height: 80px;
        height: 80px;
        width: auto;
    }
    .sideBar .Button {
        margin: 0 10%;
    }
}