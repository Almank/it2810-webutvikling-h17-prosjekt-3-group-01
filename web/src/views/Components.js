/* Exports all components such that imports in other files will be
*  cleaner and less messy. */

// Calendar components
export {AppointmentForm} from './components/calendar/AppointmentForm';
export {ContentHeader} from './components/calendar/ContentHeader';
export {DayEle} from './components/calendar/DayEle';
export {WeekCal} from './components/calendar/WeekCal';

// Note components
export {NoteContent} from './components/NoteList/NoteContent';
export {NoteLink} from './components/NoteList/NoteLink';
export {NoteList} from './components/NoteList/NoteList';

// Sidebar components
export {Button} from './components/SideBar/Button';

// Todo components
export {TodoCatelog} from './components/TodoList/TodoCatelog'
export {TodoCatelogForm} from './components/TodoList/TodoCatelogForm'
export {TodoFilter} from './components/TodoList/TodoFilter'
export {TodoForm} from './components/TodoList/TodoForm'
export {TodoList} from './components/TodoList/TodoList'
export {ToDoListItem} from './components/TodoList/TodoListItem'