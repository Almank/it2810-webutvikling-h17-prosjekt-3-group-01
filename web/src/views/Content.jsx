/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import '../assets/styles/Content.css';
import {Route} from 'react-router-dom';
import {TodoLists} from "./containers/TodoLists";

import { Notes } from "./containers/Notes";
import {Calendar} from "./containers/Calendar";


export class Content extends React.Component {
    render(){
        return (
            <div className="content">
                <Route exact path="/" render={ () => <Calendar/>}/>
                <Route path="/todolists" render={ () => <TodoLists/>}/>
                <Route path="/notebook" render={ () => <Notes/>}/>
            </div>
        );
    }
}