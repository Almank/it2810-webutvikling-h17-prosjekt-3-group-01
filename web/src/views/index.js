export * from './App.jsx';
export * from './SideBar.jsx';
