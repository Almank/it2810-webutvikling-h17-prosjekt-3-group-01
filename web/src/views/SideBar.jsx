/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import '../assets/styles/SideBar.css';
import {Button} from "./Components";


export class SideBar extends React.Component {
    render(){
        return (
            <div className="sideBar">
                <Button link="/" icon="glyphicon-home"/>
                <Button link="/todolist" icon="glyphicon-list-alt"/>
                <Button link="/notebook" icon="glyphicon-pencil"/>
            </div>
        );
    }
}
