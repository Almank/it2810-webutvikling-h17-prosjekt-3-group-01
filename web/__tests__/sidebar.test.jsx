import React from 'react';
import {SideBar} from "../src/views/SideBar";
import {Button} from "../src/views/components/SideBar/Button";

describe("SideBar", ()=> {
    it("Should render correctly", ()=> {
        const SHALLOW = shallow(<SideBar/>);
        expect(SHALLOW).toMatchSnapshot();
    })
});
describe("Button", ()=>{
    it("Should render correctly", ()=> {
        const SHALLOW = shallow(<Button link={"/"} icon={"image"}/>);
        expect(SHALLOW).toMatchSnapshot();
    })
});