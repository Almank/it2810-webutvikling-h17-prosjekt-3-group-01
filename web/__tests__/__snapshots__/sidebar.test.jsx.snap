// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render correctly 1`] = `
<Link
  className="Button"
  replace={false}
  to="/"
>
  <span
    className="glyphicon image"
  />
</Link>
`;

exports[`SideBar Should render correctly 1`] = `
<div
  className="sideBar"
>
  <Button
    icon="glyphicon-home"
    link="/"
  />
  <Button
    icon="glyphicon-list-alt"
    link="/todolist"
  />
  <Button
    icon="glyphicon-pencil"
    link="/notebook"
  />
</div>
`;
